src/diffusers/pipelines/audioldm2/pipeline_audioldm2_tts.py

```diff
@@ -0,0 +1,1002 @@
+
+        if prompt_embeds is None:
+            prompt_embeds_list = []
+            attention_mask_list = []
+
+            for tokenizer, text_encoder in zip(tokenizers, text_encoders):
+                if text_encoder.config.model_type == "clap":
+                    text_inputs = tokenizer(
+                        prompt,
+                        padding="max_length"
+                        if isinstance(tokenizer, (RobertaTokenizer, RobertaTokenizerFast))
+                        else True,
+                        max_length=tokenizer.model_max_length,
+                        truncation=True,
+                        return_tensors="pt",
+                    )
+
+                    text_input_ids = text_inputs.input_ids
+                    attention_mask = text_inputs.attention_mask
+                    untruncated_ids = tokenizer(prompt, padding="longest", return_tensors="pt").input_ids
+
+                    if untruncated_ids.shape[-1] >= text_input_ids.shape[-1] and not torch.equal(
+                        text_input_ids, untruncated_ids
+                    ):
+                        removed_text = tokenizer.batch_decode(untruncated_ids[:, tokenizer.model_max_length - 1 : -1])
+                        logger.warning(
+                            f"The following part of your input was truncated because {text_encoder.config.model_type} can "
+                            f"only handle sequences up to {tokenizer.model_max_length} tokens: {removed_text}"
+                        )
+
+                else:
+                    # Convert text to phoneme
+                    phonemes = self.text_to_phoneme(transcription)
+                    text_inputs = self.tokenizer_2(phonemes, return_tensors="pt", padding="max_length")
+                    text_input_ids = text_inputs.input_ids
+                    attention_mask = text_inputs.attention_mask
+
+                text_input_ids = text_input_ids.to(device)
+                attention_mask = attention_mask.to(device)
+
+                if text_encoder.config.model_type == "clap":
+                    prompt_embeds = text_encoder.get_text_features(
+                        text_input_ids,
+                        attention_mask=attention_mask,
+                    )
+                    # append the seq-len dim: (bs, hidden_size) -> (bs, seq_len, hidden_size)
+                    prompt_embeds = prompt_embeds[:, None, :]
+                    # make sure that we attend to this single hidden-state
+                    attention_mask = attention_mask.new_ones((batch_size, 1))
+                else:
+                    prompt_embeds = text_encoder(
+                        text_input_ids, attention_mask=attention_mask, padding_mask=attention_mask.unsqueeze(-1)
+                    )
+                    prompt_embeds = prompt_embeds[0]
+                    prompt_embeds = self.learnable_position_embedding(prompt_embeds)
```

Comment:
Do you think it would be cleaner if we added these learned position embeddings under the projection model? Since in effect we're projecting the VITS prompt embeds to another space with learned weights?